folium/features.py

```diff
@@ -1060,12 +1060,27 @@ def __init__(self, geo_data, data=None, columns=None, key_on=None,  # noqa
                 'in favor of the `bins` parameter.', DeprecationWarning)
 
         # Create color_data dict
-        if hasattr(data, 'set_index'):
+        str_color_data = None
+        if hasattr(data, 'to_dict'):
+            # This is a pd.Dataframe or pd.Series
+            # check for numbers in future color_data keys
+            future_keys = data[columns[0]] if columns is not None else data.index  # noqa
+            any_numeric_key = any([isinstance(e,
+                                   (int, float, np.floating, np.integer))
+                                   for e in future_keys])
+
+            if hasattr(data, 'set_index'):
             # This is a pd.DataFrame
-            color_data = data.set_index(columns[0])[columns[1]].to_dict()
-        elif hasattr(data, 'to_dict'):
-            # This is a pd.Series
+                data = data.set_index(columns[0])[columns[1]]
+
+            # convert the data to dictionary
```

Comment:
Nit: this comment is a bit superfluous

Reply:
Noted. I can push the change, but I'm keen to wait for the other discussion to be agreed upon and resolved